### PR TITLE
Adding section on extension aggregates.

### PR DESCRIPTION
**Pull request by gusthoff**

### CODE REVIEW

File: content/courses/advanced-ada/chapters/aggregates.rst [hunk 1]
```diff
@@ -435,14 +435,339 @@ consider using interfaces instead of null records. We'll discuss this topic
...
+type extension using a single expression. They are similar to record
+aggregates, but include the :ada:`with` keyword, and are used to write more
+complex aggregates.
 
-    - `Extension Aggregates <http://www.ada-auth.org/standards/2xrm/html/RM-4-3-2.html>`_
+Assignments to objects of derived types
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. todo::
+Before we discuss extension aggregates in more details, though, let's start
```
Comment (pat-rogers) [inline content/courses/advanced-ada/chapters/aggregates.rst:446]: details => detail

File: content/courses/advanced-ada/chapters/aggregates.rst [hunk 2]
```diff
@@ -435,14 +435,339 @@ consider using interfaces instead of null records. We'll discuss this topic
 Extension Aggregates
 --------------------
 
-.. admonition:: Relevant topics
+Extension aggregates allow us to express information for record types with
+type extension using a single expression. They are similar to record
+aggregates, but include the :ada:`with` keyword, and are used to write more
+complex aggregates.
```
Comment (pat-rogers) [inline content/courses/advanced-ada/chapters/aggregates.rst:441]: Maybe something like this:
Extension aggregates provide a convenient way to express an aggregate for a type that extends -- adds components to -- some existing type (the "ancestor"). Although mainly a mater of convenience, an extension aggregate is essential when a client wants to express an aggregate for an extension of a private ancestor type, that is, when the client doesn't have compile-time visibility to the ancestor type's components.

File: content/courses/advanced-ada/chapters/aggregates.rst [hunk 3]
```diff
@@ -435,14 +435,339 @@ consider using interfaces instead of null records. We'll discuss this topic
...
 
-.. todo::
+Before we discuss extension aggregates in more details, though, let's start
+with a simple use-case. Let's say we have:
 
-    Complete section!
+- an object :ada:`A` of tagged type :ada:`T1`, and
+
+- an object :ada:`B` of tagged type :ada:`T2`, which is derived from :ada:`T1`.
+
```
Comment (pat-rogers) [inline content/courses/advanced-ada/chapters/aggregates.rst:452]: derived from => extends 

(Depends on how precise you want to be here.)

File: content/courses/advanced-ada/chapters/aggregates.rst [hunk 4]
```diff
@@ -435,14 +435,339 @@ consider using interfaces instead of null records. We'll discuss this topic
...
+       procedure Display (P : Point_3D) is
+       begin
+          Put_Line ("(X => " & P.X'Image
+                    & ", Y => " & P.Y'Image
+                    & ", Z => " & P.Z'Image & ")");
+       end Display;
+
+    end Points;
+
+Let's now focus on the :ada:`Show_Points` procedure, where we initialize a
```
Comment (pat-rogers) [inline content/courses/advanced-ada/chapters/aggregates.rst:530]: procedure, => procedure below,

File: content/courses/advanced-ada/chapters/aggregates.rst [hunk 5]
```diff
@@ -435,14 +435,339 @@ consider using interfaces instead of null records. We'll discuss this topic
...
+
+Here, we're using :ada:`Point_1D (P_3D)` to select the :ada:`Point_1D` view of
+an object of :ada:`Point_3D` type. At this point, we have specified the
+:ada:`Point_1D` part of the aggregate, so we still have to specify the
+remaining components of the :ada:`Point_2D` type |mdash| the :ada:`Y`
+component, to be more specific. When we do that, we get the appropriate
+aggregate for the :ada:`Point_2D` type. In summary, by carefully selecting the
+appropriate view, we're able to initialize an object of ascending type
+(:ada:`Point_2D`), which contains less components, using an object of a
+descendent type (:ada:`Point_3D`), which contains more components.
```
Comment (pat-rogers) [inline content/courses/advanced-ada/chapters/aggregates.rst:771]: Really nice use of the view concept throughout!  Good work!

File: content/courses/advanced-ada/chapters/aggregates.rst (hunk 2 above)
Comment (gusthoff) [reply content/courses/advanced-ada/chapters/aggregates.rst]: I like it. I'm using this now (with minor editorial changes).

File: content/courses/advanced-ada/chapters/aggregates.rst (hunk 5 above)
Comment (gusthoff) [reply content/courses/advanced-ada/chapters/aggregates.rst:773]: Thanks! :-)

**Review by pat-rogers — APPROVED** [review]

**Pull request merged**